core/src/avm2/regexp.rs

```diff
@@ -214,12 +214,12 @@ impl<'gc> RegExp<'gc> {
     /// Implements string.replace(regex, replacement) where the replacement is
     /// a function.
     pub fn replace_fn(
-        &mut self,
+        regexp: &Object<'gc>,
```

Comment:
This change is to allow for finer-grained locking when calling back into AS code